rpc/network/db/utils.go

```diff
@@ -55,3 +59,29 @@ func FillDeprecatedURLs(network *params.Network, providers []params.RpcProvider)
 		}
 	}
 }
+
+func ExecuteWithinTransaction(db *sql.DB, fn func(tx *sql.Tx) error) (err error) {
+	tx, err := db.Begin()
+	if err != nil {
+		return fmt.Errorf("failed to begin transaction: %w", err)
+	}
+	defer func() {
+		if p := recover(); p != nil {
```

Comment:
Oh, this is an smart thing to rollback in case of a panic 👍
Perhaps we should do this for all db transactions? Or at least write it down as a guideline?